WordPress/WordPressTest/ReferrerDetailsViewModelTests.swift

```diff
@@ -0,0 +1,78 @@
+import XCTest
+@testable import WordPress
+
+class ReferrerDetailsViewModelTests: XCTestCase {
+    var sut: ReferrerDetailsViewModel!
+
+    override func setUpWithError() throws {
+        let data = StatsTotalRowData(name: "A name", data: "Some data")
+        let delegate = ViewModelDelegate()
+        sut = ReferrerDetailsViewModel(data: data, delegate: delegate)
+    }
+
+    override func tearDownWithError() throws {
+        sut = nil
+        try super.tearDownWithError()
+    }
+
+    func testUpdateTitle() {
+        XCTAssertEqual(sut.title, "A name")
+
+        let newData = StatsTotalRowData(name: "Some other name", data: "Some other data")
+        sut.update(with: newData)
+
+        XCTAssertEqual(sut.title, "Some other name")
+    }
+
+    func testSetLoadingState() {
+        XCTAssertFalse(sut.isLoading)
+
+        sut.setLoadingState(true)
+
+        XCTAssertTrue(sut.isLoading)
+    }
+
+    func testTableViewModel() {
```

Comment:
What do you think about breaking this test up into multiple tests to improve clarity?